presidential-elections/utils/model.py

```diff
@@ -449,50 +466,299 @@ def build_model(
+                "noisy_popularity",
+                aet.nnet.softmax(noisy_mu),
+                dims=("observations", "parties_complete"),
+            )
+
+            # The concentration parameter of a Dirichlet-Multinomial distribution
+            # can be interpreted as the effective number of trials.
+            #
+            # The mean (1000) is thus taken to be roughly the sample size of
+            # polls, and the standard deviation accounts for the variation in
+            # sample size.
+            concentration_polls = pm.InverseGamma(
+                "concentration_polls", mu=1000, sigma=200
             )
 
-            concentration = pm.InverseGamma("concentration", mu=1000, sigma=200)
             pm.DirichletMultinomial(
                 "N_approve",
-                a=concentration * noisy_popularity,
+                a=concentration_polls * noisy_popularity,
                 n=data_containers["observed_N"],
                 observed=data_containers["observed_polls"],
                 dims=("observations", "parties_complete"),
             )
+
+            # --------------------------------------------------------
+            #                    ELECTION RESULTS
+            #
+            # In this section we use the variables defined before to model the
+            # political families' latent popularity and how it translates into
+            # results the day of the election.
+            #
+            # Fundamental data only enter the model here; poll results enter
+            # indirectly via the latent variable and the above regression.
+            # --------------------------------------------------------
+
+            latent_mu_t0 = (
+                party_intercept
+                + election_party_intercept
+                + party_time_effect[0]
+                + election_party_time_effect[0].T
+                + aet.dot(
+                    data_containers["election_unemp"][:, None],
+                    unemployment_effect[None, :],
+                )
+            )
+            latent_mu_t0 = latent_mu_t0 + non_competing_parties["results"]
+
+            latent_pop_t0 = pm.Deterministic(
+                "latent_pop_t0",
+                aet.nnet.softmax(latent_mu_t0),
+                dims=("elections", "parties_complete"),
+            )
+
+            # The concentration parameter of a Dirichlet-Multinomial distribution
+            # can be interpreted as the effective number of trials.
+            #
+            # The mean (1000) is thus taken to be roughly the sample size of
+            # polls, and the standard deviation accounts for the variation in
+            # sample size.
+            concentration_results = pm.InverseGamma(
+                "concentration_results", mu=1000, sigma=200
+            )
             pm.DirichletMultinomial(
                 "R",
-                a=concentration * latent_pop_t0[:-1],
+                a=concentration_results * latent_pop_t0[:-1],
```

Comment:
I understand what you're doing here, and I tried that already but it really didn't work, as we don't have enough past election to really learn `concentration_results`, so the model's convergence and predictions were really bad. Did you run the model with this parametrization? Did it work?

In any case, I don't think it's a huge deal to use the same concentration for polls _and_ results, as they describe the same underlying process (and we're not using the _real_ sample size for election results anyways)

Reply:
This will need further investigation before merging, but I don't have any convergence issue with separate concentration parameters. I will have a look at these parameters' posterior distribution.

Reply:
They do!

Reply:
Nice, then we can add it! Feels nice when we solve something without realizing it 😆